tests.py

```diff
@@ -187,6 +187,20 @@ def test_prefix_names(self):
         self.m(hn.first, "vai", hn)
         self.m(hn.last, "la", hn)
 
+    def test_family_name_and_prefix(self):
+        hn = HumanName("Vincent van Gogh")
+        self.m(hn.family, "van Gogh", hn)
+        self.assertEqual(hn.family_list, [
+            [["van"], ["Gogh"]]
+        ])
+
+    def test_family_name_and_double_prefix(self):
+        hn = HumanName("Vincent van der Gogh")
+        self.m(hn.family, "van der Gogh", hn)
+        self.assertEqual(hn.family_list, [
+            [["van", "der"], ["Gogh"]],
```

Comment:
I think it makes sense to have these pieces in a separate attribute, but I'd lean more towards having separate attributes for those 2 nested lists. Maybe "affixes" and "family name", which combine together to give "last name", (which, btw, combines with middle names to give "surnames")? 

We are kind of running out of names for things. And Wikipedia [thinks](https://en.wikipedia.org/wiki/Surname) that Surname, Last name, and family name are all the same thing, so there's that. 😄

Reply:
Also, Google translate says the English translation of tussenvoegsels is "infixes". Really? There's yet another one? I did not know that was a word. (It's a good day when I can learn a new English word, so thanks!) But I guess if we're desperate, we could use that.

Reply:
Kinda makes me wonder if we could do something more useful with slices...

Reply:
Actually, that is already how slice works. Been so long since I implemented it that I forgot. It returns the concatenated string of whichever members you slice. 

So, if we added prefixes in there you could get just the last name with a slice, ex: `hn[4:4]`.

```
_members = ['title', 'first', 'middle', 'prefix', 'last', 'suffix', 'nickname']
```

Mostly what we need is just to have the prefixes in their own bucket, then we can be more flexible when we display things. Probably means changing the parse tree to add prefix in there.

Reply:
The more I dive into the surname/last name/family name, the more confused I get... There doesn't seem to be a clear definition of any of these things.

I am missing some entries in the prefixes hence the reason I made a separate list not to interfere with your work. I was looking into possible locales but that seems to be a road I really don't want to go down...

> Mostly what we need is just to have the prefixes in their own bucket, then we can be more flexible when we display things. Probably means changing the parse tree to add prefix in there.

I tried `Vincent van Gogh van Beethoven` which gave me `van Gogh van` as a middle name. I dropped that attempt for now but that is the reason I created a nested list with pairs, prefix - family name. Otherwise it would give me simply `van` twice, and (possibly) format it as ` van van Gogh Beethoven`.

Reply:
There used to be similar conversations re: "title", and "suffix" because people got tripped up on the semantic meaning of a title vs a suffix, ex "Dr" and "MD" are kinda the same but they appear in different places in the name. I think the best strategy for the parser is to focus on the position of words (name parts) in the string, and special words that join with words before/after/around them in certain conditions/parts of the name. In the future I should probably choose names that focus on that positional information and avoid semantic meaning. 

We can add some of those things from your affixes constants to prefixes. The only danger is ones that could also be names, like "fitz" and "mala".

> I tried `Vincent van Gogh van Beethoven` which gave me `van Gogh van` as a middle name

👎 That's annoying. Doesn't seem like that was my intention:

https://github.com/derek73/python-nameparser/blob/3efe1717971a52d3c05281cbc00bd300bb2898a9/nameparser/parser.py#L879-L903